game-core/src/main/java/games/strategy/triplea/ai/pro/util/ProMoveUtils.java

```diff
@@ -415,77 +403,70 @@ public static void calculateBombingRoutes(
                   u,
                   player);
         }
-        moveRoutes.add(route);
+        moves.add(new MoveDescription(unitList, route));
       }
     }
-  }
-
-  public static void doMove(
-      final List<Collection<Unit>> moveUnits,
-      final List<Route> moveRoutes,
-      final IMoveDelegate moveDel) {
-    doMove(moveUnits, moveRoutes, null, moveDel);
+    return moves;
   }
 
   /**
    * Moves the specified groups of units along the specified routes, possibly using the specified
    * transports.
    */
-  public static void doMove(
-      final List<Collection<Unit>> moveUnits,
-      final List<Route> moveRoutes,
-      final List<Map<Unit, Unit>> transportsToLoad,
-      final IMoveDelegate moveDel) {
+  public static void doMove(final List<MoveDescription> moves, final IMoveDelegate moveDel) {
 
     final GameData data = ProData.getData();
 
     // Group non-amphib units of the same type moving on the same route
-    if (transportsToLoad == null) {
-      for (int i = 0; i < moveRoutes.size(); i++) {
-        final Route r = moveRoutes.get(i);
-        for (int j = i + 1; j < moveRoutes.size(); j++) {
-          final Route r2 = moveRoutes.get(j);
-          if (r.equals(r2)) {
-            moveUnits.get(j).addAll(moveUnits.get(i));
-            moveUnits.remove(i);
-            moveRoutes.remove(i);
-            i--;
-            break;
-          }
+    // TODO: Use MoveBatcher here - or ideally at the time the moves are being generated.
+    for (int i = 0; i < moves.size(); i++) {
```

Comment:
Is this block checked by automated test? How safe is the update here?

Reply:
I don't think it is.

We should be careful to not diverge from the previous logic. I adjusted it to more closely match the previous logic, please double check. Thanks!